sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -467,6 +470,7 @@ void CurlSession::ParseChunkSize(Context const& context)
         if (this->m_chunkSize == 0)
         { // Response with no content. end of chunk
           keepPolling = false;
+          this->m_bodyStartInBuffer = index + 1;
```

Comment:
Why are we adding 1 here?

Reply:
Set the state that indicates the buffer index, after whatever was read from the buffer plus 1 to let the position to the next char.
the `index` points to the current read position. So adding one is going to the next position so the next time we read, we will get the next position.

Reply:
@vhvb1989 please submit a follow up PR with this detail in a comment.